src/Tizen.Uix.InputMethod/Tizen.Uix.InputMethod/InputMethodEditor.cs

```diff
@@ -2209,5 +2215,186 @@ public static void SendCustomGeometryUpdated()
                 throw InputMethodExceptionFactory.CreateException(error);
             }
         }
+
+        /// <summary>
+        /// Gets the selected text synchronously.
+        /// </summary>
+        /// <privilege>
+        /// http://tizen.org/privilege/ime
+        /// </privilege>
+        /// <param name="text">The selected text.</param>
+        /// <exception cref="InvalidOperationException">
+        /// This can occur due to the following reasons:
+        /// 1) The application does not have the privilege to call this function.
+        /// 2) The IME main loop has not started yet.
+        /// 3) Invalid parameter.
+        /// </exception>
+        /// <since_tizen> 6 </since_tizen>
+        public static void GetSelectedText(out string text)
+        {
+            IntPtr txt;
+            ErrorCode error = ImeGetSelectedText(out txt);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "GetSelectedText Failed with error " + error);
+                throw InputMethodExceptionFactory.CreateException(error);
+            }
+            text = Marshal.PtrToStringAnsi(txt);
+        }
+
+        /// <summary>
+        /// Called to set the prediction hint string to deliver to the input panel.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public static event EventHandler<PredictionHintSetEventArgs> PredictionHintSet
+        {
+            add
+            {
+                lock (thisLock)
+                {
+                    _imePredictionHintSetDelegate = (IntPtr predictionHint, IntPtr userData) =>
+                    {
+                        PredictionHintSetEventArgs args = new PredictionHintSetEventArgs(Marshal.PtrToStringAnsi(predictionHint));
+                        _predictionHintSet?.Invoke(null, args);
+                    };
+                    ErrorCode error = ImeEventSetPredictionHintSetCb(_imePredictionHintSetDelegate, IntPtr.Zero);
+                    if (error != ErrorCode.None)
+                    {
+                        Log.Error(LogTag, "Add PredictionHintSet Failed with error " + error);
+                    }
+                    else
+                    {
+                        _predictionHintSet += value;
+                    }
+                }
+            }
+            remove
+            {
+                lock (thisLock)
+                {
+                    _predictionHintSet -= value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Called to set the prediction hint key and result to deliver to the input panel.
```

Comment:
It is hard to understand. what is result?
For what purpose is this event used?

Reply:
The result means key value. I will change result to value.